FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Scheduler/Jobs/JobManager.cs

```diff
@@ -199,28 +214,66 @@ private async Task ExecuteJob(Job job, CancellationToken cancellationToken)
                         throw new ExecuteTaskFailedException("Task execution failed", finishedTask.Exception);
                     }
 
+                    UpdateJobTaskResult(job, finishedTask.Result);
                     tasks.Remove(finishedTask);
                 }
 
                 var context = TaskContext.Create(resourceType, job);
-                tasks.Add(Task.Run(async () => await _taskExecutor.ExecuteAsync(context, progress, cancellationToken)));
+                if (context.IsCompleted)
+                {
+                    _logger.LogInformation("Skipping completed resource '{resourceType}'.");
+                    continue;
+                }
+
+                tasks.Add(Task.Run(async () => await _taskExecutor.ExecuteAsync(context, progress, linkedTokenSource.Token)));
 
                 _logger.LogInformation("Start processing resource '{resourceType}'", resourceType);
             }
 
             try
             {
-                await Task.WhenAll(tasks);
+                var taskResults = await Task.WhenAll(tasks);
+                foreach (var taskResult in taskResults)
+                {
+                    UpdateJobTaskResult(job, taskResult);
+                }
             }
             catch (Exception ex)
             {
                 _logger.LogError("Process task failed.");
                 throw new ExecuteTaskFailedException("Task execution failed", ex);
             }
 
+            // Cancel on-going requests during execution (i.e. blob updates in async progress report).
+            executionTokenSource.Cancel();
```

Comment:
Why do we need to create a new cancellation token for this function?

Reply:
The original cancellationToken is a global token and handles outside cancellations like function stops. Here we want to cancel all on-going requests during processing a single job as the progress report is asynchronous and might be processing when all tasks were done.

Reply:
Removed the cancelling logic here.